lib/improver/utilities/mathematical_operations.py

```diff
@@ -0,0 +1,291 @@
+                        continue
+                elif self.direction_of_integration == "negative":
+                    if lower_bound <= self.end_point:
+                        continue
+            stride = np.abs(upper_bound - lower_bound)
+            upper_half_of_stride = upper_bounds_slice.data * 0.5 * stride
+            lower_half_of_stride = lower_bounds_slice.data * 0.5 * stride
+            stride_sum += lower_half_of_stride + upper_half_of_stride
+            integrated_slice.data = stride_sum
+            integrated_cubelist.append(integrated_slice.copy())
+        # Merge resulting cubes back together
+        integrated_cube = integrated_cubelist.merge_cube()
+        return integrated_cube
+
+    def process(self, cube):
+        """Integrate a specified coordinate. This is calculated by defining the
+        upper and lower bounds for the steps along a chosen coordinate
+        within the cube.
+
+        Functions utilised are:
+            1. Ensure the cube is sorted in the direction desired for
+               integration.
+            2. Prepare for integration by creating cubes that represent the
+               upper and lower limits of the integration, as well as as a
+               template cube to put the integrated output.
+            3. Perform the integration using the trapezoidal rule.
+            4. Ensure that the integrated coordinate is a dimension coordinate
+               and ensure that the integrated coordinate is sorted in the
+               desired direction.
+
+        Args:
+            cube (Iris.cube.Cube):
+                Cube containing the data to be integrated.
+
+        Returns:
+            integrated_cube (Iris.cube.Cube):
+                The cube containing the result of the integration.
+                This will contain the same metadata as the input cube.
+
+        """
+        # Make coordinate monotonic in the direction desired for integration.
+        cube = self.ensure_monotonic_increase_in_chosen_direction(cube)
+
+        upper_bounds_cube, lower_bounds_cube, integrated_cube = (
+            self.prepare_for_integration(cube))
+
+        integrated_cube = self.perform_integration(
+            upper_bounds_cube, lower_bounds_cube, integrated_cube)
+
+        # Make sure that the coordinate that has been integrated is a
+        # dimension coordinate.
+        for coord in integrated_cube.aux_coords[::-1]:
+            if coord.name() == self.coord_name_to_integrate:
+                integrated_cube = iris.util.new_axis(
+                    integrated_cube, self.coord_name_to_integrate)
+        # Make sure that the order of the coordinate that has been integrated
+        # within the integrated_cube corresponds the direction in which the
+        # cube has been integrated.
+        integrated_cube = (
+            self.ensure_monotonic_increase_in_chosen_direction(
+                integrated_cube))
+        return integrated_cube
```

Comment:
I think the returned cube should set the bounds of the integrated coordinate as well as the points. I don't think this would be difficult to do, although it also needs testing.